src/write/row_group.rs

```diff
@@ -75,6 +75,14 @@ fn compute_num_rows(columns: &[(ColumnChunk, Vec<PageWriteSpec>)]) -> Result<i64
         .unwrap_or(Ok(0))
 }
 
+#[cfg(not(feature = "bloom_filter"))]
+pub type PageIter<'a, E> = DynStreamingIterator<'a, CompressedPage, E>;
+#[cfg(feature = "bloom_filter")]
+pub type PageIter<'a, E> = (
```

Comment:
given that this is a tuple, wouldn't it be easier if `Option<&'a [u8]>` was an argument like the other arguments of the function?

Reply:
I think it made more sense to put it into the iterator because it would be collected while going through data. I can also make it `&[Option<&[u8]>]` as a separate argument but I think makes more sense to have it in the iterator